src/leader.py

```diff
@@ -152,6 +156,76 @@ def lead(self):
             self.turn(code, position)
 
 
+    def drive_in_circle(self):
+        """Drives in a circle."""
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_left()
+
+
+    def drive_wide_winding(self):
+        """Drives in a wide, snake-like pattern."""
+        self.demo_stop_flag = False
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.turn_right()
+        self.demo_turn_wait()
+
+        if self.demo_stop_flag: return
+
+        self.turn_left()
+        self.demo_turn_wait()
+
+    def drive_winding(self):
+        """Drives in a snake-like pattern."""
+        self.demo_stop_flag = False
+        self.bw.speed(MAX_SPEED)
+        self.bw.forward()
+
+        self.fw.turn(110)
+        self.demo_turn_wait()
+
+        if self.demo_stop_flag: return
+
+        self.fw.turn(70)
+        self.demo_turn_wait()
+
+
+    def stop_demo(self):
+        """
+        Grabs controller data and gives a return value based on that data.
+        Returns True if any input is given to the controller. If no input is
+        given to the controller return False.
+
+        :return: True if any input is given to the controller.
+        :rtype: Boolean
+
+        :return: False if no input is given to the controller.
+        :rtype: Boolean
+        """
+        code, position = self._controller.get_input()
+        if position is not -1:
```

Comment:
As the button is being released this if statement is going to be entered. You need to be able to check if the button pressed is the one for the demo you are in. I'm not sure how to do this without passing an argument that tells you which demo you are in and ignoring button presses from the button connected to that function